sdk/trace/simple_span_processor_test.go

```diff
@@ -168,3 +191,127 @@ func TestSimpleSpanProcessorShutdownHonorsContextCancel(t *testing.T) {
 		t.Errorf("SimpleSpanProcessor.Shutdown did not return %v, got %v", want, got)
 	}
 }
+
+func TestSimpleSpanProcessorSelfObservability(t *testing.T) {
+	tests := []struct {
+		name          string
+		enabled       bool
+		exporter      SpanExporter
+		assertMetrics func(t *testing.T, rm metricdata.ResourceMetrics)
+	}{
+		{
+			name:     "Disabled",
+			enabled:  false,
+			exporter: &simpleTestExporter{},
+			assertMetrics: func(t *testing.T, rm metricdata.ResourceMetrics) {
+				assert.Empty(t, rm.ScopeMetrics)
+			},
+		},
+		{
+			name:     "Enabled",
+			enabled:  true,
+			exporter: &simpleTestExporter{},
+			assertMetrics: func(t *testing.T, rm metricdata.ResourceMetrics) {
+				assert.Len(t, rm.ScopeMetrics, 1)
+				sm := rm.ScopeMetrics[0]
+
+				want := metricdata.ScopeMetrics{
+					Scope: instrumentation.Scope{
+						Name:      "go.opentelemetry.io/otel/sdk/trace",
+						Version:   sdk.Version(),
+						SchemaURL: semconv.SchemaURL,
+					},
+					Metrics: []metricdata.Metrics{
+						{
+							Name:        otelconv.SDKProcessorSpanProcessed{}.Name(),
+							Description: otelconv.SDKProcessorSpanProcessed{}.Description(),
+							Unit:        otelconv.SDKProcessorSpanProcessed{}.Unit(),
+							Data: metricdata.Sum[int64]{
+								DataPoints: []metricdata.DataPoint[int64]{
+									{
+										Value: 1,
+										Attributes: attribute.NewSet(
+											semconv.OTelComponentName("simple_span_processor/0"),
+											semconv.OTelComponentTypeKey.String("simple_span_processor"),
+										),
+									},
+								},
+								Temporality: metricdata.CumulativeTemporality,
+								IsMonotonic: true,
+							},
+						},
+					},
+				}
+
+				metricdatatest.AssertEqual(t, want, sm, metricdatatest.IgnoreTimestamp())
+			},
+		},
+		{
+			name:    "Enabled, Exporter error",
+			enabled: true,
+			exporter: &failingTestExporter{
+				simpleTestExporter: simpleTestExporter{},
+			},
+			assertMetrics: func(t *testing.T, rm metricdata.ResourceMetrics) {
+				assert.Len(t, rm.ScopeMetrics, 1)
+				sm := rm.ScopeMetrics[0]
+
+				want := metricdata.ScopeMetrics{
+					Scope: instrumentation.Scope{
+						Name:      "go.opentelemetry.io/otel/sdk/trace",
+						Version:   sdk.Version(),
+						SchemaURL: semconv.SchemaURL,
+					},
+					Metrics: []metricdata.Metrics{
+						{
+							Name:        otelconv.SDKProcessorSpanProcessed{}.Name(),
+							Description: otelconv.SDKProcessorSpanProcessed{}.Description(),
+							Unit:        otelconv.SDKProcessorSpanProcessed{}.Unit(),
+							Data: metricdata.Sum[int64]{
+								DataPoints: []metricdata.DataPoint[int64]{
+									{
+										Value: 1,
+										Attributes: attribute.NewSet(
+											semconv.OTelComponentName("simple_span_processor/1"),
```

Comment:
The `simpleProcessorIDCounter` should be reset for each test so this is not determined based on position of the test case, which will change (i.e. future changes or just `go test -count=2`).